src/parquet/util/comparison.h

```diff
@@ -20,40 +20,145 @@
 
 #include <algorithm>
 
+#include "parquet/exception.h"
 #include "parquet/schema.h"
 #include "parquet/types.h"
 
 namespace parquet {
 
-template <typename T>
-struct Compare {
-  explicit Compare(const ColumnDescriptor* descr) : type_length_(descr->type_length()) {}
-
-  inline bool operator()(const T& a, const T& b) { return a < b; }
+class PARQUET_EXPORT Compare {
+ public:
+  virtual ~Compare() {}
+  static std::shared_ptr<Compare> Make(const ColumnDescriptor* descr);
+};
 
- private:
-  int32_t type_length_;
+// The default comparison is SIGNED
+template <typename DType>
+class PARQUET_EXPORT CompareDefault : public Compare {
+ public:
+  typedef typename DType::c_type T;
+  CompareDefault() {}
+  virtual ~CompareDefault() {}
+  virtual bool operator()(const T& a, const T& b) { return a < b; }
```

Comment:
override (instead of virtual)

Reply:
`CompareDefault` is the base class and will require virtual. I used override inside the derived classes.